src/cryptsetup/cryptsetup.c

```diff
@@ -46,6 +46,14 @@
 #define CRYPT_SECTOR_SIZE 512
 #define CRYPT_MAX_SECTOR_SIZE 4096
 
+typedef enum PasswordRegisteredType {
```

Comment:
sounds like a misnomer, this is not about passwords really. It's about the two flavours of passphrase. hence maybe do:

```
typedef enum PassphraseType {
        PASSPHRASE_NONE,
        PASSPHRASE_REGULAR = 1 << 0,
        PASSPHRASE_RECOVERY_KEY = 1 << 1,
        PASSPHRASE_BOTH = PASSPHRASE_REGULAR|PASSPHRASE_RECOVERY_KEY,
        _PASSPHRASE_TYPE_MAX,
        _PASSPHRASE_TYPE_INVALID = -1,
} PassphraseType;
```

(as long as we only have two real types here we have the luxury we can make this a bit field without losing density, so let's do this)